### PR TITLE
feat:Provide unified metrics calculation and reporting implementation

### PR DESCRIPTION
**Pull request by gongna-au**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #433

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/metrics/collector/api_health.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/apache/dubbo-go-pixiu/pkg/metrics/global"
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (baerwang) [inline pkg/metrics/collector/api_health.go]: split import style

File: pkg/metrics/collector/api_health.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+//很多api的状态
```
Comment (baerwang) [inline pkg/metrics/collector/api_health.go]: change english

File: pkg/metrics/collector/cluster_health.go [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"path"
+
+	"github.com/apache/dubbo-go-pixiu/pkg/metrics/global"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
```
Comment (baerwang) [inline pkg/metrics/collector/cluster_health.go:14]: split import style

File: pkg/metrics/global/global.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package global
+
+//主要完成一些公共使用类代码的封装以一些全局关键字的定义。
+import ()
```
Comment (baerwang) [inline pkg/metrics/global/global.go:4]: del and change english

File: pkg/metrics/global/global.go [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+const (
+	// Exporter Namespace.
+	// Namespace将会经常被scrape和collector甚至main调用
```
Comment (baerwang) [inline pkg/metrics/global/global.go]: change english and add field meaning

File: pkg/metrics/global/global.go (hunk 4 above)
Comment (gongna-au) [reply pkg/metrics/global/global.go:4]: OK

File: pkg/metrics/collector/api_health.go [hunk 6]
```diff
@@ -0,0 +1,182 @@
+package collector
```
Comment (baerwang) [inline pkg/metrics/collector/api_health.go:1]: add license

**Comment by AlexStocks** [comment]
@gongna-au pls fix the file confliction.

File: pkg/metrics/collector/collect.go [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	)
+	scrapeSuccessDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(global.Namespace, "scrape", "success"),
+		"pixiu_exporter: Whether a collector succeeded.",
+		[]string{"collector"},
+		nil,
+	)
+)
+
+//供给 collector 调用
```
Comment (AlexStocks) [inline pkg/metrics/collector/collect.go]: english comment. do not use chinese.

File: pkg/metrics/collector/collect.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+		helpDefaultState = "enabled"
+	} else {
+		helpDefaultState = "disabled"
+	}
+
+	// Create flag for this collector
+	flagName := fmt.Sprintf("collector.%s", name)
+	flagHelp := fmt.Sprintf("Enable the %s collector (default: %s).", name, helpDefaultState)
+	defaultValue := fmt.Sprintf("%v", isDefaultEnabled)
+	//在解析和验证标志后分派给给定的函数。
```
Comment (AlexStocks) [inline pkg/metrics/collector/collect.go]: english comment, pls.

**Comment by AlexStocks** [comment]
@gongna-au pls fix the file confliction.

**Pull request closed without merging**